mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -102,21 +104,106 @@ def compute_metrics_batched(self, model):
             all_query_flattened = [
                 q for sample in self.samples for q in sample["query"]
             ]
-            all_query_embs = self._encode_unique_texts(
-                all_query_flattened, encode_corpus_func
-            )
+            if self.evaluator_type == "standard":
+                all_query_embs = self._encode_unique_texts(
+                    all_query_flattened, encode_corpus_func
+                )
+            elif self.evaluator_type == "miracl":
+                all_query_embs = np.asarray(
+                    encode_queries_func(all_query_flattened, batch_size=self.batch_size)
+                )
         else:
             raise ValueError(
                 f"Query must be a string or a list of strings but is {type(self.samples[0]['query'])}"
             )
 
+        if self.evaluator_type == "standard":
+            results = self._encode_candidates(
+                encode_corpus_func=encode_corpus_func,
+                batched=True,
+                all_query_embs=all_query_embs,
+                encode_queries_func=encode_queries_func,
+            )
+        elif self.evaluator_type == "miracl":
+            results = self._encode_candidates_miracl(all_query_embs, encode_corpus_func)
+        return results
+
+    def compute_metrics_individual(self, model):
+        """Embeds every (query, positive, negative) tuple individually.
+        Is slower than the batched version, but saves memory as only the
+        embeddings for one tuple are needed. Useful when you have
+        a really large test set
+        """
+        # using encode_queries and encode_corpus functions if they exists,
+        # which can be defined by users to add different instructions for query and passage conveniently
+        encode_queries_func = (
+            model.encode_queries if hasattr(model, "encode_queries") else model.encode
+        )
+        encode_corpus_func = (
+            model.encode_corpus if hasattr(model, "encode_corpus") else model.encode
+        )
+        if self.evaluator_type == "standard":
+            results = self._encode_candidates(
+                encode_corpus_func=encode_corpus_func,
+                encode_queries_func=encode_queries_func,
+                batched=False,
+            )
+        elif self.evaluator_type == "miracl":
+            results = self._encode_candidates_miracl_individual(
+                encode_queries_func=encode_queries_func,
+                encode_corpus_func=encode_corpus_func,
+            )
```

Comment:
I believe we can unify the encode part here?

Reply:
@KennethEnevoldsen I'm not sure where without trying to dismantle the MIRACL code.

I confirmed that the results are the same as before this MR. I improved readability, by using the same code structure for both standard and MIRACL modes. I addressed two of your suggestions—moved metric calculations to a separate method and verified encode_unique_texts is indeed ok for MIRACL—and merged the upstream changes with the task name.

Reply:
> @KennethEnevoldsen I'm not sure where without trying to dismantle the MIRACL code.

Right. Then let us keep it as is for now. It is already a great deal better than before.